src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -109,6 +109,31 @@ impl ScalarFunc {
         }
         Ok(Some(Cow::Owned(output)))
     }
+	
+    pub fn concat_ws<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let mut output_sep: Vec<u8> = Vec::new();
+        let mut output: Vec<u8> = Vec::new();
+
+        let mut index = 0;
```

Comment:
This variable actually behaves like a `bool`.
You can set the initial value to `true` and do
```rust
if thisValue {
   let input = try_opt!(expr.eval_string(ctx, row));
   output_sep.extend_from_slice(&input);
   thisValue = false;
} else {
  let input1 = try_opt!(expr.eval_string(ctx, row));
  output.extend_from_slice(&input1);
  output.extend_from_slice(&output_sep);
}
```
Which is more readable.